compiler-core/src/package_interface.rs

```diff
@@ -370,183 +375,169 @@ impl PackageInterface {
+            .filter(|(_, v)| v.publicity.is_public())
+        {
+            let _ = type_aliases.insert(
+                name.clone(),
+                TypeAliasInterface {
+                    documentation: alias.documentation.clone(),
+                    deprecation: DeprecationInterface::from_deprecation(&alias.deprecation),
+                    parameters: alias.arity,
+                    alias: TypeInterface::from_type(&alias.type_),
+                },
+            );
+        }
 
-                // A public top-level function.
-                Definition::Function(Function {
-                    publicity: Publicity::Public,
-                    name,
-                    arguments,
-                    deprecation,
-                    return_type,
-                    documentation,
-                    implementations,
-                    location: _,
-                    end_position: _,
-                    body: _,
-                    return_annotation: _,
-                    external_erlang: _,
-                    external_javascript: _,
-                }) => {
+        for (name, value) in interface
+            .values
+            .iter()
+            .filter(|(_, v)| v.publicity.is_public())
+        {
+            match (value.type_.as_ref(), value.variant.clone()) {
+                (
+                    Type::Fn {
+                        args: arguments,
+                        retrn: return_type,
+                    },
+                    ValueConstructorVariant::ModuleFn {
+                        documentation,
+                        implementations,
+                        ..
+                    },
+                ) => {
                     let mut id_map = IdMap::new();
-                    let (_, name) = name
-                        .as_ref()
-                        .expect("Function in a definition must be named");
+
                     let _ = functions.insert(
                         name.clone(),
                         FunctionInterface {
                             implementations: ImplementationsInterface::from_implementations(
-                                implementations,
+                                &implementations,
                             ),
-                            deprecation: DeprecationInterface::from_deprecation(deprecation),
-                            documentation: documentation.as_ref().map(|(_, doc)| doc.clone()),
+                            deprecation: DeprecationInterface::from_deprecation(&value.deprecation),
+                            documentation,
                             parameters: arguments
                                 .iter()
                                 .map(|arg| ParameterInterface {
-                                    label: arg.names.get_label().cloned(),
-                                    type_: from_type_helper(arg.type_.as_ref(), &mut id_map),
+                                    //TODO: Find label
```

Comment:
Is this unfinished?

Reply:
Oh, yeah, forgot about this part. I couldn't find how to retrieve the parameter label from the `ModuleInterface`. Maybe the arguments in the `Fn` variant of `Type` needs to be converted from a `Vec<Arc<Type>>` to a ``Vec<(Option<EcoString>, Arc<Type>)>`?